File: 2021/day02/day02.swift
#!/usr/bin/env swift

import Foundation

// Advent of Code 2021
// Day 02

let content = try! String(contentsOfFile: "input.txt", encoding: .utf8)
let lines = content.components(separatedBy: "\n").filter { !$0.isEmpty }
let strings = lines.map { $0.split(separator: " ") }
let commands = strings.map { ($0[0], Int($0[1]) ?? 0) }

// Part One:
// What do you get if you multiply your final horizontal position by your final depth?
var position = 0
var depth = 0

for c in commands {
    switch c.0 {
        case "down":
            depth += c.1
        case "up":
            depth -= c.1

        case "forward":
            position = position + c.1

        default:
            print("Unknown command: \(c.0)")
    }
}

print("Part One:", position * depth)

// Part Two:
// What do you get if you multiply your final horizontal position by your final depth?
var aim = 0
position = 0
depth = 0

for c in commands {
    switch c.0 {
        case "down":
            aim += c.1
        case "up":
            aim -= c.1
        case "forward":
            position += c.1
            depth += c.1 * aim

        default:
            print("Unknown command: \(c.0)")
    }
}

print("Part Two:", position * depth)

File: 2021/day02/day02.py
#!/usr/bin/python3
import itertools as itt

# # Advent of Code 2021
# Day 02

file = open('input.txt')
lines = [line.strip() for line in file]

if __name__ == '__main__':

    # Part One:
    # What do you get if you multiply your final horizontal position by your final depth?
    horizontal = 0
    depth = 0
    for l in lines:
        if l.startswith("down"):
            depth += int(l[5:])
        if l.startswith("up"):
            depth -= int(l[3:])

        if l.startswith("forward"):
            horizontal += int(l[8:])

    print( "Part One: {:d}".format(horizontal * depth) )

    # Part Two:
    # What do you get if you multiply your final horizontal position by your final depth?
    horizontal = 0
    depth = 0
    aim = 0
    for l in lines:
        if l.startswith("down"):
            aim += int(l[5:])
        if l.startswith("up"):
            aim -= int(l[3:])

        if l.startswith("forward"):
            target = int(l[8:])
            horizontal += target
            depth += target * aim

    print( "Part Two: {:d}".format(horizontal * depth) )
